{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 61,
      "metadata": {},
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Python 3.10.11\n",
            "Requirement already satisfied: pip in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (24.1)\n"
          ]
        }
      ],
      "source": [
        "!python --version\n",
        "!python -m pip install --upgrade pip\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 62,
      "metadata": {
        "id": "7jtLyvRXlrmz"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: absl-py==1.4.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 1)) (1.4.0)\n",
            "Requirement already satisfied: asttokens==2.2.1 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 2)) (2.2.1)\n",
            "Requirement already satisfied: backcall==0.2.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 3)) (0.2.0)\n",
            "Requirement already satisfied: boto3==1.26.111 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 4)) (1.26.111)\n",
            "Requirement already satisfied: botocore==1.29.111 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 5)) (1.29.111)\n",
            "Requirement already satisfied: cachetools==5.3.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 6)) (5.3.0)\n",
            "Requirement already satisfied: certifi==2022.12.7 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 7)) (2022.12.7)\n",
            "Requirement already satisfied: chardet==4.0.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 8)) (4.0.0)\n",
            "Requirement already satisfied: charset-normalizer==3.1.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 9)) (3.1.0)\n",
            "Requirement already satisfied: click==8.0.4 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 10)) (8.0.4)\n",
            "Requirement already satisfied: colorama==0.4.6 in c:\\users\\maoto\\appdata\\roaming\\python\\python310\\site-packages (from -r requirements.txt (line 11)) (0.4.6)\n",
            "Requirement already satisfied: contourpy==1.0.7 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 12)) (1.0.7)\n",
            "Requirement already satisfied: cycler==0.10.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 13)) (0.10.0)\n",
            "Requirement already satisfied: decorator==5.1.1 in c:\\users\\maoto\\appdata\\roaming\\python\\python310\\site-packages (from -r requirements.txt (line 14)) (5.1.1)\n",
            "Requirement already satisfied: executing==1.2.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 15)) (1.2.0)\n",
            "Collecting filelock==3.11.0 (from -r requirements.txt (line 16))\n",
            "  Using cached filelock-3.11.0-py3-none-any.whl.metadata (2.5 kB)\n",
            "Requirement already satisfied: fire==0.5.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 17)) (0.5.0)\n",
            "Requirement already satisfied: fonttools==4.39.3 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 18)) (4.39.3)\n",
            "Requirement already satisfied: gitdb==4.0.10 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 19)) (4.0.10)\n",
            "Requirement already satisfied: GitPython==3.1.31 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 20)) (3.1.31)\n",
            "Requirement already satisfied: google-auth==2.17.2 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 21)) (2.17.2)\n",
            "Requirement already satisfied: google-auth-oauthlib==1.0.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 22)) (1.0.0)\n",
            "Requirement already satisfied: grpcio==1.53.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 23)) (1.53.0)\n",
            "Requirement already satisfied: huggingface-hub==0.13.4 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 24)) (0.13.4)\n",
            "Requirement already satisfied: idna==2.10 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 25)) (2.10)\n",
            "Requirement already satisfied: ipython==8.12.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 26)) (8.12.0)\n",
            "Requirement already satisfied: jedi==0.18.2 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 27)) (0.18.2)\n",
            "Collecting Jinja2==3.1.2 (from -r requirements.txt (line 28))\n",
            "  Using cached Jinja2-3.1.2-py3-none-any.whl.metadata (3.5 kB)\n",
            "Requirement already satisfied: jmespath==1.0.1 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 29)) (1.0.1)\n",
            "Requirement already satisfied: kiwisolver==1.4.4 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 30)) (1.4.4)\n",
            "Requirement already satisfied: Markdown==3.4.3 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 31)) (3.4.3)\n",
            "Collecting MarkupSafe==2.1.2 (from -r requirements.txt (line 32))\n",
            "  Using cached MarkupSafe-2.1.2-cp310-cp310-win_amd64.whl.metadata (3.3 kB)\n",
            "Requirement already satisfied: matplotlib==3.7.1 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 33)) (3.7.1)\n",
            "Requirement already satisfied: matplotlib-inline==0.1.6 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 34)) (0.1.6)\n",
            "Requirement already satisfied: mpmath==1.3.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 35)) (1.3.0)\n",
            "Collecting networkx==3.1 (from -r requirements.txt (line 36))\n",
            "  Using cached networkx-3.1-py3-none-any.whl.metadata (5.3 kB)\n",
            "Requirement already satisfied: nnAudio==0.3.2 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 37)) (0.3.2)\n",
            "Collecting numpy==1.24.2 (from -r requirements.txt (line 38))\n",
            "  Using cached numpy-1.24.2-cp310-cp310-win_amd64.whl.metadata (5.6 kB)\n",
            "Requirement already satisfied: oauthlib==3.2.2 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 39)) (3.2.2)\n",
            "Collecting opencv-python==4.7.0.72 (from -r requirements.txt (line 40))\n",
            "  Using cached opencv_python-4.7.0.72-cp37-abi3-win_amd64.whl.metadata (18 kB)\n",
            "Requirement already satisfied: packaging==23.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 41)) (23.0)\n",
            "Collecting pandas==2.0.0 (from -r requirements.txt (line 42))\n",
            "  Using cached pandas-2.0.0-cp310-cp310-win_amd64.whl.metadata (18 kB)\n",
            "Requirement already satisfied: parso==0.8.3 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 43)) (0.8.3)\n",
            "Requirement already satisfied: pickleshare==0.7.5 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 44)) (0.7.5)\n",
            "Collecting Pillow==9.5.0 (from -r requirements.txt (line 45))\n",
            "  Using cached Pillow-9.5.0-cp310-cp310-win_amd64.whl.metadata (9.7 kB)\n",
            "Requirement already satisfied: prompt-toolkit==3.0.38 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 46)) (3.0.38)\n",
            "Requirement already satisfied: protobuf==4.22.1 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 47)) (4.22.1)\n",
            "Requirement already satisfied: psutil==5.9.4 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 48)) (5.9.4)\n",
            "Requirement already satisfied: pure-eval==0.2.2 in c:\\users\\maoto\\appdata\\roaming\\python\\python310\\site-packages (from -r requirements.txt (line 49)) (0.2.2)\n",
            "Requirement already satisfied: pyasn1==0.4.8 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 50)) (0.4.8)\n",
            "Requirement already satisfied: pyasn1-modules==0.2.8 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 51)) (0.2.8)\n",
            "Requirement already satisfied: pybboxes==0.1.6 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 52)) (0.1.6)\n",
            "Requirement already satisfied: pydub==0.25.1 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 53)) (0.25.1)\n",
            "Collecting Pygments==2.15.0 (from -r requirements.txt (line 54))\n",
            "  Using cached Pygments-2.15.0-py3-none-any.whl.metadata (2.5 kB)\n",
            "Requirement already satisfied: pyparsing==2.4.7 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 55)) (2.4.7)\n",
            "Collecting python-dateutil==2.8.2 (from -r requirements.txt (line 56))\n",
            "  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl.metadata (8.2 kB)\n",
            "Requirement already satisfied: python-dotenv==1.0.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 57)) (1.0.0)\n",
            "Collecting pytz==2023.3 (from -r requirements.txt (line 58))\n",
            "  Using cached pytz-2023.3-py2.py3-none-any.whl.metadata (22 kB)\n",
            "Requirement already satisfied: PyYAML==6.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 59)) (6.0)\n",
            "Collecting requests==2.28.2 (from -r requirements.txt (line 60))\n",
            "  Using cached requests-2.28.2-py3-none-any.whl.metadata (4.6 kB)\n",
            "Requirement already satisfied: requests-oauthlib==1.3.1 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 61)) (1.3.1)\n",
            "Requirement already satisfied: requests-toolbelt==0.10.1 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 62)) (0.10.1)\n",
            "Requirement already satisfied: roboflow==1.0.3 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 63)) (1.0.3)\n",
            "Requirement already satisfied: rsa==4.9 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 64)) (4.9)\n",
            "Requirement already satisfied: s3transfer==0.6.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 65)) (0.6.0)\n",
            "Requirement already satisfied: sahi==0.11.13 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 66)) (0.11.13)\n",
            "Collecting scipy==1.10.1 (from -r requirements.txt (line 67))\n",
            "  Using cached scipy-1.10.1-cp310-cp310-win_amd64.whl.metadata (58 kB)\n",
            "Requirement already satisfied: seaborn==0.12.2 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 68)) (0.12.2)\n",
            "Requirement already satisfied: shapely==2.0.1 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 69)) (2.0.1)\n",
            "Requirement already satisfied: six==1.16.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 70)) (1.16.0)\n",
            "Requirement already satisfied: smmap==5.0.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 71)) (5.0.0)\n",
            "Requirement already satisfied: stack-data==0.6.2 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 72)) (0.6.2)\n",
            "Collecting sympy==1.11.1 (from -r requirements.txt (line 73))\n",
            "  Using cached sympy-1.11.1-py3-none-any.whl.metadata (12 kB)\n",
            "Requirement already satisfied: tensorboard==2.12.1 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 74)) (2.12.1)\n",
            "Requirement already satisfied: tensorboard-data-server==0.7.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 75)) (0.7.0)\n",
            "Requirement already satisfied: tensorboard-plugin-wit==1.8.1 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 76)) (1.8.1)\n",
            "Requirement already satisfied: termcolor==2.2.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 77)) (2.2.0)\n",
            "Requirement already satisfied: terminaltables==3.1.10 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 78)) (3.1.10)\n",
            "Requirement already satisfied: thop==0.1.1.post2209072238 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 79)) (0.1.1.post2209072238)\n",
            "Collecting torch==2.0.0 (from -r requirements.txt (line 80))\n",
            "  Using cached torch-2.0.0-cp310-cp310-win_amd64.whl.metadata (23 kB)\n",
            "Collecting torchaudio==2.0.1 (from -r requirements.txt (line 81))\n",
            "  Using cached torchaudio-2.0.1-cp310-cp310-win_amd64.whl.metadata (1.2 kB)\n",
            "Collecting torchvision==0.15.1 (from -r requirements.txt (line 82))\n",
            "  Using cached torchvision-0.15.1-cp310-cp310-win_amd64.whl.metadata (11 kB)\n",
            "Requirement already satisfied: tqdm==4.65.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 83)) (4.65.0)\n",
            "Requirement already satisfied: traitlets==5.9.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 84)) (5.9.0)\n",
            "Collecting typing_extensions==4.5.0 (from -r requirements.txt (line 85))\n",
            "  Using cached typing_extensions-4.5.0-py3-none-any.whl.metadata (8.5 kB)\n",
            "Collecting tzdata==2023.3 (from -r requirements.txt (line 86))\n",
            "  Using cached tzdata-2023.3-py2.py3-none-any.whl.metadata (1.4 kB)\n",
            "Requirement already satisfied: urllib3==1.26.15 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 87)) (1.26.15)\n",
            "Requirement already satisfied: wcwidth==0.2.6 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 88)) (0.2.6)\n",
            "Requirement already satisfied: Werkzeug==2.2.3 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 89)) (2.2.3)\n",
            "Requirement already satisfied: wget==3.2 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 90)) (3.2)\n",
            "Requirement already satisfied: yolov5==7.0.11 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from -r requirements.txt (line 91)) (7.0.11)\n",
            "Requirement already satisfied: setuptools>=41.0.0 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from tensorboard==2.12.1->-r requirements.txt (line 74)) (67.0.0)\n",
            "Requirement already satisfied: wheel>=0.26 in c:\\users\\maoto\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from tensorboard==2.12.1->-r requirements.txt (line 74)) (0.41.1)\n",
            "Using cached filelock-3.11.0-py3-none-any.whl (10.0 kB)\n",
            "Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)\n",
            "Using cached MarkupSafe-2.1.2-cp310-cp310-win_amd64.whl (16 kB)\n",
            "Using cached networkx-3.1-py3-none-any.whl (2.1 MB)\n",
            "Using cached numpy-1.24.2-cp310-cp310-win_amd64.whl (14.8 MB)\n",
            "Using cached opencv_python-4.7.0.72-cp37-abi3-win_amd64.whl (38.2 MB)\n",
            "Using cached pandas-2.0.0-cp310-cp310-win_amd64.whl (11.2 MB)\n",
            "Using cached Pillow-9.5.0-cp310-cp310-win_amd64.whl (2.5 MB)\n",
            "Using cached Pygments-2.15.0-py3-none-any.whl (1.1 MB)\n",
            "Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)\n",
            "Using cached pytz-2023.3-py2.py3-none-any.whl (502 kB)\n",
            "Using cached requests-2.28.2-py3-none-any.whl (62 kB)\n",
            "Using cached scipy-1.10.1-cp310-cp310-win_amd64.whl (42.5 MB)\n",
            "Using cached sympy-1.11.1-py3-none-any.whl (6.5 MB)\n",
            "Using cached torch-2.0.0-cp310-cp310-win_amd64.whl (172.3 MB)\n",
            "Using cached torchaudio-2.0.1-cp310-cp310-win_amd64.whl (2.1 MB)\n",
            "Using cached torchvision-0.15.1-cp310-cp310-win_amd64.whl (1.2 MB)\n",
            "Using cached typing_extensions-4.5.0-py3-none-any.whl (27 kB)\n",
            "Using cached tzdata-2023.3-py2.py3-none-any.whl (341 kB)\n",
            "Installing collected packages: pytz, tzdata, typing_extensions, sympy, requests, python-dateutil, Pygments, Pillow, numpy, networkx, MarkupSafe, filelock, scipy, pandas, opencv-python, Jinja2, torch, torchvision, torchaudio\n",
            "  Attempting uninstall: pytz\n",
            "    Found existing installation: pytz 2024.1\n",
            "    Uninstalling pytz-2024.1:\n",
            "      Successfully uninstalled pytz-2024.1\n",
            "  Attempting uninstall: tzdata\n",
            "    Found existing installation: tzdata 2024.1\n",
            "    Uninstalling tzdata-2024.1:\n",
            "      Successfully uninstalled tzdata-2024.1\n",
            "  Attempting uninstall: typing_extensions\n",
            "    Found existing installation: typing_extensions 4.12.2\n",
            "    Uninstalling typing_extensions-4.12.2:\n",
            "      Successfully uninstalled typing_extensions-4.12.2\n",
            "  Attempting uninstall: sympy\n",
            "    Found existing installation: sympy 1.12.1\n",
            "    Uninstalling sympy-1.12.1:\n",
            "      Successfully uninstalled sympy-1.12.1\n",
            "  Attempting uninstall: requests\n",
            "    Found existing installation: requests 2.32.3\n",
            "    Uninstalling requests-2.32.3:\n",
            "      Successfully uninstalled requests-2.32.3\n",
            "  Attempting uninstall: python-dateutil\n",
            "    Found existing installation: python-dateutil 2.9.0.post0\n",
            "    Uninstalling python-dateutil-2.9.0.post0:\n",
            "      Successfully uninstalled python-dateutil-2.9.0.post0\n",
            "  Attempting uninstall: Pygments\n",
            "    Found existing installation: Pygments 2.18.0\n",
            "    Uninstalling Pygments-2.18.0:\n",
            "      Successfully uninstalled Pygments-2.18.0\n",
            "  Attempting uninstall: Pillow\n",
            "    Found existing installation: pillow 10.3.0\n",
            "    Uninstalling pillow-10.3.0:\n",
            "      Successfully uninstalled pillow-10.3.0\n",
            "  Attempting uninstall: numpy\n",
            "    Found existing installation: numpy 1.26.4\n",
            "    Uninstalling numpy-1.26.4:\n",
            "      Successfully uninstalled numpy-1.26.4\n",
            "  Attempting uninstall: networkx\n",
            "    Found existing installation: networkx 3.3\n",
            "    Uninstalling networkx-3.3:\n",
            "      Successfully uninstalled networkx-3.3\n",
            "  Attempting uninstall: MarkupSafe\n",
            "    Found existing installation: MarkupSafe 2.1.5\n",
            "    Uninstalling MarkupSafe-2.1.5:\n",
            "      Successfully uninstalled MarkupSafe-2.1.5\n",
            "  Attempting uninstall: filelock\n",
            "    Found existing installation: filelock 3.15.3\n",
            "    Uninstalling filelock-3.15.3:\n",
            "      Successfully uninstalled filelock-3.15.3\n",
            "  Attempting uninstall: scipy\n",
            "    Found existing installation: scipy 1.13.1\n",
            "    Uninstalling scipy-1.13.1:\n",
            "      Successfully uninstalled scipy-1.13.1\n",
            "  Attempting uninstall: pandas\n",
            "    Found existing installation: pandas 2.2.2\n",
            "    Uninstalling pandas-2.2.2:\n",
            "      Successfully uninstalled pandas-2.2.2\n",
            "  Attempting uninstall: opencv-python\n",
            "    Found existing installation: opencv-python 4.8.1.78\n",
            "    Uninstalling opencv-python-4.8.1.78:\n",
            "      Successfully uninstalled opencv-python-4.8.1.78\n",
            "  Attempting uninstall: Jinja2\n",
            "    Found existing installation: Jinja2 3.1.4\n",
            "    Uninstalling Jinja2-3.1.4:\n",
            "      Successfully uninstalled Jinja2-3.1.4\n",
            "  Attempting uninstall: torch\n",
            "    Found existing installation: torch 2.3.1\n",
            "    Uninstalling torch-2.3.1:\n",
            "      Successfully uninstalled torch-2.3.1\n",
            "  Attempting uninstall: torchvision\n",
            "    Found existing installation: torchvision 0.18.1\n",
            "    Uninstalling torchvision-0.18.1:\n",
            "      Successfully uninstalled torchvision-0.18.1\n",
            "  Attempting uninstall: torchaudio\n",
            "    Found existing installation: torchaudio 2.3.1\n",
            "    Uninstalling torchaudio-2.3.1:\n",
            "      Successfully uninstalled torchaudio-2.3.1\n",
            "Successfully installed Jinja2-3.1.2 MarkupSafe-2.1.2 Pillow-9.5.0 Pygments-2.15.0 filelock-3.11.0 networkx-3.1 numpy-1.24.2 opencv-python-4.7.0.72 pandas-2.0.0 python-dateutil-2.8.2 pytz-2023.3 requests-2.28.2 scipy-1.10.1 sympy-1.11.1 torch-2.0.0 torchaudio-2.0.1 torchvision-0.15.1 typing_extensions-4.5.0 tzdata-2023.3\n"
          ]
        }
      ],
      "source": [
        "!pip install -r requirements.txt\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 63,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NLYVrLtYt0Y5",
        "outputId": "30a18f48-ae24-4531-8389-dd5a8097be63"
      },
      "outputs": [
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "\n",
            "A module that was compiled using NumPy 1.x cannot be run in\n",
            "NumPy 2.0.0 as it may crash. To support both 1.x and 2.x\n",
            "versions of NumPy, modules must be compiled with NumPy 2.0.\n",
            "Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.\n",
            "\n",
            "If you are a user of the module, the easiest solution will be to\n",
            "downgrade to 'numpy<2' or try to upgrade the affected module.\n",
            "We expect that some modules will need time to support NumPy 2.\n",
            "\n",
            "Traceback (most recent call last):  File \"c:\\Users\\maoto\\AppData\\Local\\Programs\\Python\\Python310\\lib\\runpy.py\", line 196, in _run_module_as_main\n",
            "    return _run_code(code, main_globals, None,\n",
            "  File \"c:\\Users\\maoto\\AppData\\Local\\Programs\\Python\\Python310\\lib\\runpy.py\", line 86, in _run_code\n",
            "    exec(code, run_globals)\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\ipykernel_launcher.py\", line 18, in <module>\n",
            "    app.launch_new_instance()\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\traitlets\\config\\application.py\", line 1075, in launch_instance\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\ipykernel\\kernelapp.py\", line 739, in start\n",
            "    self.io_loop.start()\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\tornado\\platform\\asyncio.py\", line 205, in start\n",
            "    self.asyncio_loop.run_forever()\n",
            "  File \"c:\\Users\\maoto\\AppData\\Local\\Programs\\Python\\Python310\\lib\\asyncio\\base_events.py\", line 603, in run_forever\n",
            "    self._run_once()\n",
            "  File \"c:\\Users\\maoto\\AppData\\Local\\Programs\\Python\\Python310\\lib\\asyncio\\base_events.py\", line 1909, in _run_once\n",
            "    handle._run()\n",
            "  File \"c:\\Users\\maoto\\AppData\\Local\\Programs\\Python\\Python310\\lib\\asyncio\\events.py\", line 80, in _run\n",
            "    self._context.run(self._callback, *self._args)\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\ipykernel\\kernelbase.py\", line 545, in dispatch_queue\n",
            "    await self.process_one()\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\ipykernel\\kernelbase.py\", line 534, in process_one\n",
            "    await dispatch(*args)\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\ipykernel\\kernelbase.py\", line 437, in dispatch_shell\n",
            "    await result\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\ipykernel\\ipkernel.py\", line 362, in execute_request\n",
            "    await super().execute_request(stream, ident, parent)\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\ipykernel\\kernelbase.py\", line 778, in execute_request\n",
            "    reply_content = await reply_content\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\ipykernel\\ipkernel.py\", line 449, in do_execute\n",
            "    res = shell.run_cell(\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\ipykernel\\zmqshell.py\", line 549, in run_cell\n",
            "    return super().run_cell(*args, **kwargs)\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\interactiveshell.py\", line 3075, in run_cell\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\interactiveshell.py\", line 3130, in _run_cell\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\async_helpers.py\", line 129, in _pseudo_sync_runner\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\interactiveshell.py\", line 3334, in run_cell_async\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\interactiveshell.py\", line 3517, in run_ast_nodes\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\interactiveshell.py\", line 3577, in run_code\n",
            "  File \"C:\\Users\\maoto\\AppData\\Local\\Temp\\ipykernel_22476\\1772531124.py\", line 1, in <module>\n",
            "    import pandas as pd, silic\n",
            "  File \"a:\\dev\\project\\silic\\silic.py\", line 2, in <module>\n",
            "    import numpy as np, pandas as pd, torch, cv2, os, time, shutil, sys, argparse, mimetypes\n"
          ]
        },
        {
          "ename": "AttributeError",
          "evalue": "_ARRAY_API not found",
          "output_type": "error",
          "traceback": [
            "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
            "\u001b[1;31mAttributeError\u001b[0m: _ARRAY_API not found"
          ]
        },
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Unexpected exception formatting exception. Falling back to standard exception\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Traceback (most recent call last):\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\interactiveshell.py\", line 3577, in run_code\n",
            "  File \"C:\\Users\\maoto\\AppData\\Local\\Temp\\ipykernel_22476\\1772531124.py\", line 1, in <module>\n",
            "    import pandas as pd, silic\n",
            "  File \"a:\\dev\\project\\silic\\silic.py\", line 2, in <module>\n",
            "    import numpy as np, pandas as pd, torch, cv2, os, time, shutil, sys, argparse, mimetypes\n",
            "ImportError: numpy.core.multiarray failed to import\n",
            "\n",
            "During handling of the above exception, another exception occurred:\n",
            "\n",
            "Traceback (most recent call last):\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\pygments\\styles\\__init__.py\", line 45, in get_style_by_name\n",
            "ModuleNotFoundError: No module named 'pygments.styles.default'\n",
            "\n",
            "During handling of the above exception, another exception occurred:\n",
            "\n",
            "Traceback (most recent call last):\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\interactiveshell.py\", line 2168, in showtraceback\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\ultratb.py\", line 1454, in structured_traceback\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\ultratb.py\", line 1345, in structured_traceback\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\ultratb.py\", line 1192, in structured_traceback\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\IPython\\core\\ultratb.py\", line 1133, in get_records\n",
            "  File \"C:\\Users\\maoto\\AppData\\Roaming\\Python\\Python310\\site-packages\\pygments\\styles\\__init__.py\", line 47, in get_style_by_name\n",
            "pygments.util.ClassNotFound: Could not find style module 'pygments.styles.default', though it should be builtin.\n"
          ]
        }
      ],
      "source": [
        "import pandas as pd, silic\n",
        "from IPython.display import Image"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 64,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6Ch3SLzzuIVw",
        "outputId": "2a4b3205-ad89-4f44-bf74-c1da81a5cc5a"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "STFT kernels created, time used = 0.0471 seconds\n",
            "STFT kernels created, time used = 0.0494 seconds\n",
            "STFT filter created, time used = 0.0055 seconds\n",
            "Mel filter created, time used = 0.0055 seconds\n",
            "Origional audio: channel = 1, sample_rate = 32000 Hz, sample_size = 192000, duration = 6.0 s\n",
            "Standarized audio: channel = 1, sample_rate = 32000 Hz, sample_size = 192000, duration = 6.0 s\n",
            "Spectrogram was saved to A:/dev/project/s1/新增資料夾\\linear\\49593.png.\n",
            "Spectrogram was saved to A:/dev/project/s1/新增資料夾\\rainbow\\49593.png.\n",
            "Origional audio: channel = 2, sample_rate = 44100 Hz, sample_size = 1388544, duration = 15.74312925170068 s\n",
            "Standarized audio: channel = 1, sample_rate = 32000 Hz, sample_size = 503780, duration = 15.743125 s\n",
            "Spectrogram was saved to A:/dev/project/s1/新增資料夾\\linear\\SILIC-demo.png.\n",
            "Spectrogram was saved to A:/dev/project/s1/新增資料夾\\rainbow\\SILIC-demo.png.\n",
            "Origional audio: channel = 2, sample_rate = 44100 Hz, sample_size = 5291520, duration = 59.99455782312925 s\n",
            "Standarized audio: channel = 1, sample_rate = 32000 Hz, sample_size = 1919826, duration = 59.9945625 s\n",
            "Spectrogram was saved to A:/dev/project/s1/新增資料夾\\linear\\TT01_20210401_060000.png.\n",
            "Spectrogram was saved to A:/dev/project/s1/新增資料夾\\rainbow\\TT01_20210401_060000.png.\n",
            "Origional audio: channel = 2, sample_rate = 44100 Hz, sample_size = 5291520, duration = 59.99455782312925 s\n",
            "Standarized audio: channel = 1, sample_rate = 32000 Hz, sample_size = 1919826, duration = 59.9945625 s\n",
            "Spectrogram was saved to A:/dev/project/s1/新增資料夾\\linear\\ZZG01_20210501_070000.png.\n",
            "Spectrogram was saved to A:/dev/project/s1/新增資料夾\\rainbow\\ZZG01_20210501_070000.png.\n"
          ]
        },
        {
          "name": "stderr",
          "output_type": "stream",
          "text": [
            "Fusing layers... \n",
            "YOLOv5s summary: 214 layers, 8014822 parameters, 0 gradients, 19.1 GFLOPs\n"
          ]
        }
      ],
      "source": [
        "!python silic-ui.py"
      ]
    }
  ],
  "metadata": {
    "accelerator": "GPU",
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.10.11"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}
